pkg/resources/flow.go

```diff
@@ -55,13 +52,16 @@ func (flowTemplateFactory) Kind() string {
 func (flowTemplateFactory) New(def client.ResourceDefinition, c client.Interface, gc interfaces.GraphContext) interfaces.Resource {
 	newFlow := parametrizeResource(def.Flow, gc).(*client.Flow)
 
+	dep := gc.Dependency()
+	depName := strings.Replace(dep.Name, dep.GenerateName, "", 1)
+
 	return report.SimpleReporter{
 		BaseResource: &Flow{
 			Base:          Base{def.Meta},
 			flow:          newFlow,
 			context:       gc,
 			status:        interfaces.ResourceNotReady,
-			generatedName: fmt.Sprintf("%s-%v", newFlow.Name, atomic.AddUint32(&counter, 1)),
+			generatedName: fmt.Sprintf("%s-%s%s", newFlow.Name, depName, gc.GetArg("AC_NAME")),
```

Comment:
both parts are generated, so there is no need for the dash. Also there are limitations on resource name length